packages/customize-widgets/src/components/inserter/use-inserter.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect, useRef } from '@wordpress/element';
+import { focus } from '@wordpress/dom';
+import { ESCAPE } from '@wordpress/keycodes';
+
+const {
+	wp: { customize },
+} = window;
+
+export const inserterId = 'widgets-inserter';
+
+const OuterSection = customize.OuterSection;
+// Override the OuterSection class to handle multiple outer sections.
+// It closes all the other outer sections whenever one is opened.
+// The result is that at most one outer section can be opened at the same time.
+customize.OuterSection = class extends OuterSection {
+	onChangeExpanded( expanded, args ) {
+		if ( expanded ) {
+			customize.section.each( ( section ) => {
+				if (
+					section.params.type === 'outer' &&
+					section.id !== this.id
+				) {
+					if ( section.expanded() ) {
+						section.collapse();
+					}
+				}
+			} );
+		}
+
+		return super.onChangeExpanded( expanded, args );
+	}
+};
+// Handle constructor so that "params.type" can be correctly pointed to "outer".
+customize.sectionConstructor.outer = customize.OuterSection;
```

Comment:
When this feature is merged to Core, should these changes be made in `wp-customize` instead of patched in here? It might be worth noting that in a comment.

Reply:
Yeah, could be! (Just to be clear, when you say `wp-customize` do you mean the customize module (`js/_enqueues/wp/customize/`) in Core?)

Reply:
Yes exactly.

Reply:
What do you think about moving this bit of code that re-declares `OuterSection` to `customize-sidebar-constructors.js`? I'm just really worried that this will be missed when merging to Core 🙂